lib/request.js

```diff
@@ -242,9 +243,12 @@ module.exports = {
    */
 
   get hostname() {
```

Comment:
This needs some adjusting - the code path is weird and for some reason I'm not using `host` whilst also assigning `const host = this.host` (relic from the past). I'll clean this up later today.